cdk/cdk.sh

```diff
@@ -1,56 +1,103 @@
+            
+            pushd "cdk-resources/${path}" || exit
+            npx projen new awscdk-construct --npm-access "public" --author "MongoDBAtlas" --author-name "MongoDBAtlas" --docgen true --sample-code false --name '@mongodbatlas-awscdk/'"${path}" --author-address 'https://mongodb.com' --cdk-version '2.1.0' --default-release-branch 'master' --major-version 1 --release-to-npm true --repository-url 'https://github.com/mongodb/mongodbatlas-cloudformation-resources.git' --description 'Retrieves or creates '"${path}"' in any given Atlas organization' --keywords {'cdk','awscdk','aws-cdk','cloudformation','cfn','extensions','constructs','cfn-resources','cloudformation-registry','l1','mongodb','atlas',"$path"}
             rm -rf .git
-            cd ../.. || exit
+            popd
           fi
       done
-done
+fi
+
+if [ "$cdk_type" = "L2" ]; then
+  echo "Generating L2 CDK resource"
+  mkdir "l2-cdk-resources/${resource}"
+  pushd "l2-cdk-resources/${resource}"
+
+  npx projen new awscdk-construct --npm-access "public" --author "MongoDBAtlas" --author-name "MongoDBAtlas" --docgen true --sample-code false --name "@mongodbatlas-awscdk-l2/${resource}" --author-address 'https://mongodb.com' --cdk-version '2.1.0' --default-release-branch 'master' --major-version 1 --release-to-npm true --repository-url 'https://github.com/mongodb/mongodbatlas-cloudformation-resources.git' --description "Retrieves or creates ${resource} in any given Atlas organization" --keywords {'cdk','awscdk','aws-cdk','cloudformation','cfn','extensions','constructs','cfn-resources','cloudformation-registry','l2','mongodb','atlas',"$resource"}
+  rm -rf .git
+
+  popd
+fi
+
+if [ "$cdk_type" = "L3" ]; then
+  echo "Generating L3 CDK resource"
+  mkdir "l3-cdk-resources/${resource}"
+  pushd "l3-cdk-resources/${resource}"
+
+  npx projen new awscdk-construct --npm-access "public" --author "MongoDBAtlas" --author-name "MongoDBAtlas" --docgen true --sample-code false --name "@mongodbatlas-awscdk-l3/${resource}" --author-address 'https://mongodb.com' --cdk-version '2.1.0' --default-release-branch 'master' --major-version 1 --release-to-npm true --repository-url 'https://github.com/mongodb/mongodbatlas-cloudformation-resources.git' --description "Retrieves or creates ${resource} in any given Atlas organization" --keywords {'cdk','awscdk','aws-cdk','cloudformation','cfn','extensions','constructs','cfn-resources','cloudformation-registry','l3','mongodb','atlas',"$resource"}
+  rm -rf .git
+
+  popd
+fi
```

Comment:
this code seems to be almost identical (the only diff is l1/l2/l3) with 2 more sections above.
Can we extract common code into a function?
Example: https://github.com/10gen/mms/blob/43a6f3534ad8c23591b2a063e25462296c2abac7/scripts/evergreen/openapi/save_openapi_files.sh#L18

Reply:
Updated